interfaces/system.go

```diff
@@ -0,0 +1,31 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package interfaces
+
+// IsTheSystemSnap returns true if snapName is one of the possible
+// names for the snap representing the system.
+func IsTheSystemSnap(snapName string) bool {
+	switch snapName {
+	case "", "snapd", "core", "ubuntu-core":
```

Comment:
is `""` for implicit plugs/slots? Aren't those always attached to either core or snapd?